pkg/apis/core/types.go

```diff
@@ -3316,6 +3316,14 @@ type ServiceSpec struct {
 	// of peer discovery.
 	// +optional
 	PublishNotReadyAddresses bool
+
+	// topologyKeys is a preference-order list of topology keys which clients should respect when accessing this Service.
+	// If any ready backends exist for index [0], they should always be chosen; only if no backends exist for index [0] should backends for index [1] be considered.
+	// If this field is specified, but all entries have no backends that match the topology of the client, the service has no backends for that client and connections should fail.
+	// The special value "" may be used to means "any node". This catch-all value, if used, only makes sense as the last value in the list.
+	// If this is not specified or empty, no topoly contraints will be applied.
```

Comment:
s/topoly/topology

Reply:
Do we want to add something like "At most 8 keys may be specified" ?  We historically have not set limits on list fields, but we probably should have.

Reply:
Can you explain why should we limit it ?

Reply:
That make sense, how about limit to 16 ?